core/dbt/adapters/base/impl.py

```diff
@@ -1518,12 +1608,14 @@ def render_model_constraint(cls, constraint: ModelLevelConstraint) -> Optional[s
         else:
             return None
 
-    @classmethod
-    def has_feature(cls, feature: AdapterFeature) -> bool:
-        # The base adapter implementation does not implement any optional
-        # features, so always return false. Adapters which wish to provide
-        # optional features will have to override this function.
-        return False
+    def capabilities(self) -> List[CapabilitySupport]:
```

Comment:
What do you think about having this method a `Dict[Capability, CapabilitySupport]` (perhaps wrapping it in our own type with a default lookup) instead of a `List[CapabilitySupport]`? That way, adapters just need to implement this method and `capability_support` could do a lookup on `capabilities`

This assumes that capabilities won't be costly to call, and will largely look like returning a static dictionary as you've done for the postgres implemention

Reply:
Implemented this suggestion.